analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -82,45 +83,62 @@ public abstract class UtilityAnalyzerBase<TSyntaxKind, TMessage> : UtilityAnalyz
 
         protected virtual bool AnalyzeUnchangedFiles => false;
 
-        protected virtual IEnumerable<TMessage> CreateAnalysisMessages(CompilationAnalysisContext c) => Enumerable.Empty<TMessage>();
+        protected virtual IEnumerable<TMessage> CreateAnalysisMessages(Compilation compilation) => Enumerable.Empty<TMessage>();
 
         protected UtilityAnalyzerBase(string diagnosticId, string title) : base(diagnosticId, title) { }
 
         protected sealed override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterCompilationAction(c =>
+            context.RegisterCompilationStartAction(startContext =>
+            {
+                ReadParameters(context, startContext.Options, startContext.Compilation);
+                if (!IsAnalyzerEnabled)
+                {
+                    return;
+                }
+
+                ConcurrentStack<TMessage> treeMessages = new();
```

Comment:
Why does this need a concurrency on a non-concurrent analyzer?

Reply:
for two reasons:
* I propose to enable concurrent execution: https://github.com/SonarSource/sonar-dotnet/pull/6576#pullrequestreview-1226434176 I did a test run while doing the performance comparison with concurrency enabled.
* The delegate passed in `RegisterSemanticModelAction` might be called concurrently even though `EnableConcurrentExecution => false`. I'm not sure about the behavior and this way we are on the safe side.

Reply:
Let's revert this until we really need this. The concurrency is a topic for another PR.